sdk/java/src/main/java/sawtooth/sdk/processor/State.java

```diff
@@ -144,5 +148,57 @@ public State(Stream stream, String contextId) {
 
     return addressesThatWereSet;
   }
+  
+  /**Make an add event request on something happend on State Object for a specific context. 
+   * 
+   * @param eventType {String} - This is used to subscribe to events. It should
+   *     be globally unique and describe what, in general, has occurred
+   * @param attributes {Collection{key,value}} - Additional information about the event that
+   *     is transparent to the validator.  Attributes can be used by subscribers to
+   *     filter the type of events they receive   
+   * @param data {ByteString} - Additional information about the event that is
+   *     opaque to the validator 
+   * @throws InternalError something went wrong while processing transaction
+   */
 
+  public void addEvent(String eventType,
+                       Collection<java.util.Map.Entry<String, String>> 
+                       attributes,ByteString data)
+                       throws InternalError, InvalidTransactionException {    
+    if (attributes == null) {
+      attributes = new ArrayList<>();
+    }
+    sawtooth.sdk.protobuf.Event.Attribute.Builder attributeBuilder = null;
```

Comment:
Shouldn't have to fully qualify this, given the import.  It can just be:

`Event.Attribute.Builder attributeBuilder = null`